lighthouse-core/gather/gatherers/js-usage.js

```diff
@@ -34,9 +43,23 @@ class JsUsage extends Gatherer {
     /** @type {Record<string, Array<LH.Crdp.Profiler.ScriptCoverage>>} */
     const usageByUrl = {};
     for (const scriptUsage of scriptUsages) {
-      const scripts = usageByUrl[scriptUsage.url] || [];
+      // ScriptCoverages without a URL are from code we run over the protocol.
+      // We shouldn't analyze ourselves, so skip.
+      if (scriptUsage.url === '') continue;
+
+      // `ScriptCoverage.url` can be overridden by a magic sourceURL comment.
+      // Lookup the url via the scriptId, if possible.
+      let scriptUsageUrl = scriptUsage.url;
+
+      // ...
+      // TODO how to map script id to url ...
+
+      // `scriptUsage.url` can sometimes be relative to the final url, so normalize to an absolute
+      // url by using the URL ctor.
+      const url = new URL(scriptUsageUrl, finalUrl).href;
```

Comment:
i have a suspicion this baseUrl isn't what we want... we really want the URL of the frame that this script is executing in.

(being in a 3rd party iframe would mess this up)


this obv makes the implementation a bit more complicated. :(

Reply:
I actually should remove this, its only relative when its essentially random user input.